### PR TITLE
virt UEFI fix

### LINKED ISSUE #59188: virt: efi=True forces update of VM
**Description**

When applying a state like the following, the VM should be defined and started at the first `state.apply`, but the subsequent ones should do nothing.

```yaml
vm01:
  virt.running:
    - name: vm01
    - cpu: 1
    - mem: 512
    - boot:
        efi: True
    - disks:
      - name: system
        pool: default
        image: salt://SLES15-SP2-JeOS.x86_64-15.2-kvm-and-xen-GM.qcow2
    - seed: False
```

**Steps to Reproduce the behavior**

Apply 2 times the above state. The second time, the return states `definition: True` indicating the VM definition has been updated.

**Expected behavior**

The second time, the return should state `definition: False` since the state hasn't changed.

### PR DESCRIPTION
**Pull request by cbosdo**
### What does this PR do?

Don't report an change if user provides `efi=True` and we already have a `loader` set in the XML definition of the VM.

The issue comes from the fact that libvirt transforms the `firmware='efi'` into a pair of `<loader>` and `<nvram>` elements with the automatically picked paths. When updating the VM if `efi=True` we need to assume the already defined loader and nvram are unchanged.

The PR also converts one of the `virt.update()` tests to pytest and split it into smaller more manageable test functions.

### What issues does this PR fix or reference?
Fixes: #59188

### Merge requirements satisfied?
- [ ] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by cbosdo** [comment]
rerun pr-ubuntu1804-py3-pytest

**Comment by cbosdo** [comment]
@waynew I just rebased the PR on top of master after the network PR merge. You're good to review that one. Hopefully it shouldn't be that hard even given the loc changes: most of it is unit test moving/splitting.

File: tests/pytests/unit/modules/virt/test_helpers.py
Comment (s0undt3ch) [inline tests/pytests/unit/modules/virt/test_helpers.py:39]: Could we covert this name away from camel case? We want to dissociate these tests from unittest `assert<Something>`

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/virt/test_helpers.py
Comment (cbosdo) [reply tests/pytests/unit/modules/virt/test_helpers.py:39]: sure, I'll change those

Comment (cbosdo) [reply tests/pytests/unit/modules/virt/test_helpers.py]: Done. I also added a commit to change the same thing for the remaining calls in the `tests/unit/modules/test_virt.py` file to ease future tests migration.

File: salt/modules/virt.py
Comment (waynew) [inline salt/modules/virt.py:4191]: **question**: I'm guessing this is correct - just wanted to double check that `"serial"` and `"console"` should only be plural (have an S) in the list passed to `_skip_update`, correct? I see that it's not being updated in the `to_skip` dict - just the singular.

Is that expected?

File: salt/modules/virt.py
Comment (waynew) [inline salt/modules/virt.py:2011]: **suggestion (non-blocking)**: I'd love if we could refactor this function a bit. The bool logic smells funny (`is False` and `is True` is a code smell), and we can do `parent_tag.find("loader")` up to 3 times in here.

I took a stab on the test clinic to try and come up with a reasonable refactor but I'm not sure I grok enough about it.

Anyway, I don't think it should block this PR at all, but it'd be awesome if it could get simplified at some point!

**Review by waynew — DISMISSED** [review]
@cbosdo Thanks for the PR! It looks great - checked this out locally and ran the tests, +1 for migrating to pytests.

I left a question - if I understood correctly then this is :100: 

I also left a suggestion but I don't think there's any reason to hold this up.

File: salt/modules/virt.py
Comment (cbosdo) [reply salt/modules/virt.py:4191]: that's correct. The array contains `virt.update()` parameter names (which are plural), while the key is used in the `_diff_*_lists()` name and are singular.

File: salt/modules/virt.py
Comment (cbosdo) [reply salt/modules/virt.py:2011]: I agree this one would need a refactoring. I'll try to come to it later, but not in this PR.

**Comment by cbosdo** [comment]
re-run all

**Comment by cbosdo** [comment]
re-run archlts-py3-pytest

**Comment by cbosdo** [comment]
re-run centos7-py3-tcp-pytest

**Comment by cbosdo** [comment]
re-run centos8-py3-pytest

**Comment by cbosdo** [comment]
re-run debian9-py3-pytest

**Comment by cbosdo** [comment]
re-run debian10-py3-pytest

**Comment by cbosdo** [comment]
re-run amazon2-py3-pytest

**Comment by cbosdo** [comment]
re-run debian9-py3-pytest

**Comment by cbosdo** [comment]
re-run centos7-py3-pytest

**Comment by cbosdo** [comment]
re-run opensuse15-py3-pytest

**Comment by cbosdo** [comment]
re-run ubuntu1604-py3-tcp-pytest

**Comment by cbosdo** [comment]
re-run ubuntu1804-py3-pytest

File: tests/pytests/unit/modules/virt/test_domain.py
Comment (s0undt3ch) [inline tests/pytests/unit/modules/virt/test_domain.py:1090]: For improved diff readability of what pytest reports when there are failures, the asserts should be in the form of:
```python
assert what_we_got == what_we_expect
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/virt/test_domain.py
Comment (s0undt3ch) [reply tests/pytests/unit/modules/virt/test_domain.py:1090]: The above applies to almost all of the newly added asserts

Comment (cbosdo) [reply tests/pytests/unit/modules/virt/test_domain.py:1090]: oh, that was never clear to me which way it was... and I think we had both ways depending on the code. I'll fix those today

Comment (cbosdo) [reply tests/pytests/unit/modules/virt/test_domain.py]: @s0undt3ch I changed the ones introduced in this PR. To ease patches handing, I will change the other asserts in the virt pytests once the opened PRs are merged.

Comment (s0undt3ch) [reply tests/pytests/unit/modules/virt/test_domain.py]: Sounds good

Comment (s0undt3ch) [reply tests/pytests/unit/modules/virt/test_domain.py]: Thank You.

**Review by s0undt3ch — APPROVED** [review]

**Comment by cbosdo** [comment]
re-run all

**Comment by cbosdo** [comment]
re-run ubuntu1604-py3-pycryptodome-pytest

**Comment by cbosdo** [comment]
re-run macosxcatalina-py3-pytest

**Pull request merged**
